internal/uplink/metrics.go

```diff
@@ -0,0 +1,17 @@
+package uplink
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+)
+
+var (
+	uc = promauto.NewHistogramVec(prometheus.HistogramOpts{
```

Comment:
I think a counter value is better, like: https://github.com/brocaar/chirpstack-network-server/blob/master/internal/backend/gateway/mqtt/metrics.go#L9

If there is a downlink (data) response, then this histogram will include the duration of the `get_downlink_data_delay` + the downlink processing. If there isn't a downlink response, then the uplink function will return early. Therefore I don't think using a histogram over a counter doesn't give you any additional information that can be used to tune the performance.

E.g. it could be useful to add around database queries, but that would be for a different PR :)

Reply:
alright